delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/routes/BatchFilesRoutes.scala

```diff
@@ -0,0 +1,108 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.storage.files.routes
+
+import akka.http.scaladsl.model.StatusCodes.Created
+import akka.http.scaladsl.server._
+import cats.data.EitherT
+import cats.effect.IO
+import cats.syntax.all._
+import ch.epfl.bluebrain.nexus.delta.kernel.Logger
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.batch.BatchFiles
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.FileRejection._
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.{CopyFileDestination, File, FileRejection}
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.permissions.{read => Read}
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.{contexts, FileResource}
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.StoragesConfig.StorageTypeConfig
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.{ContextValue, RemoteContextResolution}
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.encoder.JsonLdEncoder
+import ch.epfl.bluebrain.nexus.delta.rdf.utils.JsonKeyOrdering
+import ch.epfl.bluebrain.nexus.delta.sdk._
+import ch.epfl.bluebrain.nexus.delta.sdk.acls.AclCheck
+import ch.epfl.bluebrain.nexus.delta.sdk.circe.CirceUnmarshalling
+import ch.epfl.bluebrain.nexus.delta.sdk.directives.DeltaDirectives._
+import ch.epfl.bluebrain.nexus.delta.sdk.directives.{AuthDirectives, DeltaSchemeDirectives}
+import ch.epfl.bluebrain.nexus.delta.sdk.error.ServiceError.AuthorizationFailed
+import ch.epfl.bluebrain.nexus.delta.sdk.identities.Identities
+import ch.epfl.bluebrain.nexus.delta.sdk.identities.model.Caller
+import ch.epfl.bluebrain.nexus.delta.sdk.jsonld.BulkOperationResults
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{BaseUri, IdSegment}
+import kamon.instrumentation.akka.http.TracingDirectives.operationName
+
+final class BatchFilesRoutes(
+    identities: Identities,
+    aclCheck: AclCheck,
+    batchFiles: BatchFiles,
+    schemeDirectives: DeltaSchemeDirectives,
+    index: IndexingAction.Execute[File]
+)(implicit
+    baseUri: BaseUri,
+    storageConfig: StorageTypeConfig,
+    cr: RemoteContextResolution,
+    ordering: JsonKeyOrdering
+) extends AuthDirectives(identities, aclCheck)
+    with CirceUnmarshalling { self =>
+
+  private val logger = Logger[BatchFilesRoutes]
+
+  import baseUri.prefixSegment
+  import schemeDirectives.resolveProjectRef
+
+  implicit val bulkOpJsonLdEnc: JsonLdEncoder[BulkOperationResults[FileResource]] =
+    BulkOperationResults.searchResultsJsonLdEncoder(ContextValue(contexts.files))
+
+  def routes: Route =
+    baseUriPrefix(baseUri.prefix) {
+      pathPrefix("bulk") {
+        pathPrefix("files") {
+          extractCaller { implicit caller =>
+            resolveProjectRef.apply { projectRef =>
+              (post & pathEndOrSingleSlash & parameter("storage".as[IdSegment].?) & indexingMode & tagParam) {
+                (storage, mode, tag) =>
+                  operationName(s"$prefixSegment/files/{org}/{project}") {
```

Comment:
Not needed, they still remain in some places but in the end it is disabled as in some cases:
*  this could not properly override the default behaviour (which is recording metrics for the given url)
* Some urls like the ones with a resource id for example were introducing a cardinality that Prometheus could not handle
* Prometheus would crash when there were too many resource reads
* Delta would also eventually crash as the collected metrics would continue to pile up until an OOM